client/js/components/support/DpSupport.vue

```diff
@@ -77,7 +77,14 @@ export default {
     }),
 
     visibleContacts () {
-      return Object.values(this.contacts).filter(contact => contact.attributes.visible)
+      const contacts = []
+      Object.values(this.contacts).forEach(contact => {
+        if (contact.attributes.visible !== false) {
+          contacts.push(contact)
+        }
+      })
+
+      return contacts
```

Comment:
but this code (below) returns an array containing the contacts for which the visible attribute is either true or **not  set** (undefined). So it could work for users which don't have the permission to see the _visible_ flag

```
Object.values(this.contacts).forEach(contact => {
        if (contact.attributes.visible !== false) {
          contacts.push(contact)
        }
      })
```

Reply:
true, but I find it quite opaque to see when it is set and when it is not. It suggests a check for a boolean.

Reply:
sorry, what do you mean by that - _It suggests a check for a boolean._

Reply:
so you suggest to add a filter to the fetchContacts-Request?

Reply:
this looks like a boolean--Check for me: `(contact.attributes.visible !== false)` but it also checks "undefined" as a known state of the attribute. That is - in my opinion - not good readable, if You don't know the architecture behind that logic. -> when its undefined and when its false.

Reply:
> so you suggest to add a filter to the fetchContacts-Request?
Yes

Reply:
As I understand, _onlyVisible_ is a condition name, any unique name can be used here, right?

But If I add it like this → 

```
mounted () {
    let params = {
      fields: {
        CustomerContact: [
          'title',
          'phoneNumber',
          'text',
          'eMailAddress',
          'visible'
        ].join()
      }
    }
    if (hasPermission('feature_customer_support_contact_administration')) {
      params = {
        ...params,
        filter: {
          onlyVisible: {
            condition: {
              path: 'visible',
              value: true
            }
          }
        }
      }
    }

    this.fetchContacts(params)
  }
```
an error get an error --> Faulty API request: No property 'visible' is available in the type class 'demosplan\DemosPlanCoreBundle\ResourceTypes\CustomerContactResourceType'. Available properties are:

Reply:
by resource type --> if ($this->hasManagementPermission()) {
            $properties[] = $this->createAttribute($this->visible)->readable()->initializable();
        }

Reply:
True. There has to be added a  `->filterable()`

Reply:
done here https://github.com/demos-europe/demosplan-core/pull/2300/commits/c597514d0f14e2bec06b8c98cef103f0f70ccbd0